src/sketch.js

```diff
@@ -31,111 +34,219 @@
-// should not be called prior to storing activeCreatureIndex in setup()
-function getActiveCreature() {
-  let activeCreatureIndex = getItem("activeCreatureIndex");
-  if (!activeCreatureIndex) {
-    storeItem("activeCreatureIndex", 0);
-    activeCreatureIndex = 0;
+  get width() {
+    return document.documentElement.clientWidth;
   }
-  const activeCreatureName = [...CREATURES.keys()][activeCreatureIndex];
-  activeCreature = CREATURES.get(activeCreatureName);
-  return activeCreature, activeCreatureName;
+
+  get height() {
+    return document.documentElement.clientHeight;
+  }
+
+  get currentX() {
+    return this.#activeCreature.x;
+  }
+
+  set currentX(x) {
+    this.#activeCreature.x = x;
+    storeItem(this.#activeCreature.name + "X", x);
+  }
+
+  get currentY() {
+    return this.#activeCreature.y;
+  }
+
+  set currentY(y) {
+    this.#activeCreature.y = y;
+    storeItem(this.#activeCreature.name + "Y", y);
+  }
+
+  // smallAxis is the size of the axis that is smaller than screenBreakpoint
+  // if multiple exist, it returns the smaller axis
+  // if none exist, it returns null
+  get smallAxis() {
+    return this.width < this.screenBreakpoint ||
+      this.height < this.screenBreakpoint
+      ? Math.min(this.width, this.height)
+      : null;
+  }
+
+  // if screen is small, resize creatures and position accordingly
+  get creatureSize() {
+    return this.smallAxis
+      ? this.#unscaledCreatureSize -
+          0.1 * (this.screenBreakpoint - this.smallAxis)
+      : this.#unscaledCreatureSize;
+  }
+
+  get arrowSize() {
+    return this.creatureSize / 4;
+  }
+}
+
+/**
+ * creates a p5.Image object and attaches creature metadata
+ *
+ * TODO collect more member info and attach here
+ *
+ * @returns {p5.Image}
+ */
+const newCreature = (path, name, humanName) => {
+  const img = loadImage(path);
+  img.name = name;
+  img.humanName = humanName;
+  return img;
+};
+
+let sesh;
+function preload() {
+  const arrow = loadImage("/assets/arrow.svg");
+  const panda = newCreature("/assets/panda.svg", "panda", "selene");
+  const bunny = newCreature("/assets/bunny.svg", "bunny", "lucy");
+  const jellyfish = newCreature("/assets/jellyfish.svg", "jellyfish", "julie");
+  const frog = newCreature("/assets/frog.svg", "frog", "yen");
+  /** 𖡼.𖤣𖥧𖡼.𖤣𖥧 STEP 1: ADD CREATURE ABOVE THIS COMMENT 𖡼.𖤣𖥧𖡼.𖤣𖥧 */
+
+  sesh = new CreatureSession([panda, bunny, jellyfish, frog], arrow);
+  /** 𖡼.𖤣𖥧𖡼.𖤣𖥧 STEP 2: ADD CREATURE ABOVE THIS COMMENT 𖡼.𖤣𖥧𖡼.𖤣𖥧 */
 }
 
-function setActiveCreature(activeCreatureIndex) {
-  if (activeCreatureIndex == null) {
-    throw new Error("Please specify an activeCreatureIndex!!!");
+function setup() {
+  let previouslyLoaded = localStorage.getItem("previouslyLoaded");
+  if (!previouslyLoaded) {
+    // first time loaded! initialize creature coordinates.
+    localStorage.setItem("previouslyLoaded", "1");
+    sesh.initialize();
+  } else {
+    // get creature coordinates from localStorage
+    sesh.reinitialize();
```

Comment:
woah woah woah so creature position will persist??? cool

Reply:
ohh yea! its been persisting >:)

Reply:
i mainly added it cuz the script would reload and start from the beginning when you went to a new page